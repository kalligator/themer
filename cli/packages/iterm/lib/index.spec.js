const { render, renderInstructions } = require('./index'),
  { colors } = require('../../colors-default');

describe('iTerm theme generator', () => {
  const promisedFiles = render(colors, {});

  it('should render two files when given a color set with both a light and a dark theme', async () => {
    const files = await Promise.all(promisedFiles);
    expect(files.length).toBe(2);
    expect(files.some((file) => /dark/.test(file.name))).toBe(true);
    expect(files.some((file) => /light/.test(file.name))).toBe(true);
  });

  it('should render well-formed files without missing values', async () => {
    const files = await Promise.all(promisedFiles);
    expect(
      files.some((file) => /undefined/.test(file.contents.toString('utf8'))),
    ).toBe(false);
  });

  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
