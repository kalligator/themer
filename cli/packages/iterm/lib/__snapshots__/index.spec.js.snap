// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iTerm theme generator should provide installation instructions 1`] = `
"
1. Launch iTerm
2. Press \`command\`-\`I\` to open the iTerm preferences
3. Choose Colors > Color Presets... > Import... and choose the generated theme file (\`themer-iterm-dark.itermcolors\` or \`themer-iterm-light.itermcolors\`)
"
`;
